tests/integration/security/testdata/plaintext-to-permissive.yaml

```diff
@@ -0,0 +1,21 @@
+apiVersion: security.istio.io/v1beta1
+kind: PeerAuthentication
+metadata:
+  name: default
+  annotations:
+    test-suite: plaintext-to-permissive
+spec:
+  mtls:
+    mode: PERMISSIVE
```

Comment:
PERMISSIVE means `Connection can be either plaintext or mTLS tunnel.` (https://istio.io/latest/docs/reference/config/security/peer_authentication/#PeerAuthentication-MutualTLS-Mode), do we expect it to also accept simple TLS? or should we actually use `DISABLE` in this case?

cc @incfly

Reply:
I think we should, otherwise we break applications when they are injected which is violating our goal of "transparent" proxy

Reply:
Also we previously did

Reply:
Let me  make sure I understand this correctly:
Case 1. DestinationRule sets the tls mode to DISABLE and the client envoy will use plaintext but the connection is actually simple TLS because the client application is making request to HTTPS port. This doesn't work with PeerAuthN.Permissive as the permissive mode only works with plaintext and mTLS.

Case 2. DestinationRule sets the tls mode to SIMPLE and the client envoy will use simple TLS, this doesn't work with PeerAuthN permissive for the same reason.

I agree it makes sense to support simple TLS in permissive mode. we should also update the istio.io documentation for the change.

It is probably a "bug" that it worked previously because the API indicates that the permissive is for plaintext or mTLS, not including simple TLS. and we also do not have e2e tests covering this before.

Also curious does the simple TLS work with PeerAuthN.Disable mode?

Reply:
Yes, your described Case 1/2 are correct. Also the client can have no sidecar and send a TLS request (essentially the same as case 1, but this could happen with 0 Istio CRs deployed)

> the API indicates that the permissive is for plaintext or mTLS

I suspect the documentation was just poorly phrased and the intended behave was always that it was "mtls or not mtls", not "mtls or plaintext", but perhaps that is not the case. We should update the docs to indicate this, if we are in agreement the current doc is not right.

> Also curious does the simple TLS work with PeerAuthN.Disable mode?

Yes, unless you use perPort settings (documented in the issue)

Reply:
Ah,yeah, the reason to use istio ALPN as match is because we want to support case we describe here.

Question, I think if we have the echo app serves https traffic and let a naked client send HTTPS traffic under PERMISSIVE, then this should catch this bug right?

Reply:
Yes, that test is added here. This specific test (plaintext-to-permissive.yaml) is to be more exhaustive, the bug would be caught by the other test changes (to send HTTPS requests) on its own.

Reply:
NVM, just saw this PR includes that test. Great!

Reply:
just want to confirm the root cause:
In 1.7, the FC has no `raw_buffer` nor `tls`, simple TLS worked because Envoy injects the `tls_inspector` implicitly in this case
In 1.8, the FC has the `raw_buffer` for https://github.com/envoyproxy/envoy/issues/13601 but not `tls`, and this caused Envoy to stop injecting the tls_inspector implicitly which then caused the issue here
And this PR creates a new FC with the `tls` transport protocol explicitly? Does this rely on the fact that envoy injects the `tls_inspector` automaticaly?

Reply:
1.7 chains
```
{"destinationPort":7070}
{"destinationPort":7070,"transportProtocol":"tls","applicationProtocols":["istio-peer-exchange","istio"]}
```

1.8 chains
```
{"destinationPort":7070,"transportProtocol":"raw_buffer"}
{"destinationPort":7070,"transportProtocol":"tls","applicationProtocols":["istio-peer-exchange","istio"]}
```

This PR
```
{"destinationPort":7070,"transportProtocol":"raw_buffer"}
{"destinationPort":7070,"transportProtocol":"tls"}
{"destinationPort":7070,"transportProtocol":"tls","applicationProtocols":["istio-peer-exchange","istio"]}
```

The reason we added raw_buffer at all is because of https://github.com/envoyproxy/envoy/issues/13601. Basically, we want the tls inspector only on *some* ports. But if we don't set the `transportProtocol` explicitly on all ports, envoy automatically injects and over-ambitious tls_inspector which applies to *all* ports. To get around that, we added the explicit raw_buffer, but this broke plain TLS.

Reply:
That's good summary thanks. I saw the Envoy bug is now closed does this PR is to fix the old proxy without the Envoy fix? Eventually when Envoy fix gets to all the version we support, does it mean it's safe to go back to 1.7 versions? It seems like current PR would still work, but less could be clearer.

Reply:
I suspect we could go back to 1.7 chains in Istio master, but would need to test it. The change just got merged ~today. Unless we backport to 1.8 envoy, we still need this at least for the short term